addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1801,6 +1801,225 @@ QUnit.test('composer should be focused automatically after clicking on the send
     );
 });
 
+QUnit.test('handle error when message fetching failed and there is no loaded message yet', async function (assert) {
+    assert.expect(3);
+
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        name: "General",
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                throw new Error();
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    assert.containsNone(
+        document.body,
+        '.o_MessageList',
+        "should not show message list"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed',
+        "should show loading error message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed .btn',
+        "should show a button to allow user to retry loading"
+    );
+});
+
+QUnit.test('retry button shown when message fetching failed loads messages when clicked', async function (assert) {
+    assert.expect(3);
+
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        name: "General",
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                throw new Error();
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    assert.containsNone(
+        document.body,
+        '.o_MessageList',
+        "should not show message list"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed',
+        "should show loading error message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed .btn',
+        "should show a button to allow user to retry loading"
+    );
+});
+
+QUnit.test('handle error when message fetching failed and there are loaded messages', async function (assert) {
```

Comment:
_"failure on loading more messages should display error and prompt retry button"_